packages/datadog-plugin-azure-service-bus/src/producer.js

```diff
@@ -47,7 +52,8 @@ class AzureServiceBusProducerPlugin extends ProducerPlugin {
       if (isBatch) {
         span.setTag('messaging.batch.message_count', messages.count)
         if (batchLinksAreEnabled()) {
-          messages._spanContexts.forEach(spanContext => {
+          const contexts = spanContexts.get(messages) || []
+          contexts.forEach(spanContext => {
             span.addLink(spanContext)
           })
```

Comment:
We always prefer `for...of` instead of `forEach`:

```suggestion
          const contexts = spanContexts.get(messages)
          if (contexts) {
            for (const spanContext of contexts) {
              span.addLink(spanContext)
            }
          }
```